packages/components/src/Stack.js

```diff
@@ -0,0 +1,60 @@
+import React from 'react'
+import { Box } from './Box'
+
+export const merge = (...objs) => {
+  return objs.reduce(function(merged, currentValue) {
+    return mergeTwo(merged, currentValue)
+  }, {})
+}
+
+const createSpace = (direction, space) => {
+  if (direction === 'vertical') {
+    return {
+      marginBottom: space,
+      marginRight: 0,
+    }
+  }
+
+  return {
+    marginBottom: 0,
+    marginRight: space,
+  }
+}
+
+export const Stack = React.forwardRef(
+  (
+    {
+      space = 2,
+      direction = 'vertical',
+      inline = false,
+      justify = 'normal',
+      align = 'normal',
+      children,
+      ...props
+    },
+    ref
+  ) => {
+    const styles = {
+      display: inline ? 'inline-flex' : 'flex',
+      width: '100%',
+      justifyContent: justify,
+      alignItems: align,
+    }
+
+    if (Array.isArray(direction)) {
+      styles.flexDirection = direction.map(d =>
+        d === 'vertical' ? 'column' : 'row'
+      )
+      styles['> *:not(:last-child)'] = direction.map(d => createSpace(d, space))
+    } else {
+      styles.flexDirection = direction === 'vertical' ? 'column' : 'row'
+      styles['> *:not(:last-child)'] = createSpace(direction, space)
```

Comment:
I think it would be better to use the lobotomized owl selector (`> * + *`) instead here so it would not override any margin defined on the children.

[Here's an example.](https://codesandbox.io/s/infallible-proskuriakova-5zwzg)

Reply:
I disagree, I think adding any complex selectors (either this current implementation or the lobotomized owl) here is dangerous to be honest.

A lot of Theme UI is based on the `Box` primitive and anything like this is going to lead to specificty issues

Reply:
Something simimlar to what Braid does could be done, they add a bottom spacing to each item (using padding), and add a negative margin using the same spacing to the container.

Reference:
* https://github.com/seek-oss/braid-design-system/blob/master/lib/components/Stack/Stack.tsx#L116
* https://github.com/seek-oss/braid-design-system/blob/master/lib/components/Stack/Stack.tsx#L31

And you can inspect an example on [their playroom](https://seek-oss.github.io/braid-design-system/playroom/#?code=N4Igxg9gJgpiBcIA8BlALgQzAawAQGcAHLGAXgB0R8BbDAGzsoD5yA7XXJABTpIAsIdWACdcfGAEsA5nzSlgAFgAMAX1wB6Fu049%2BgkWMky5i1Rq0duvMDAFCYo8dNnzlazWyTr0WbFpAqQA)

Reply:
This is the approach I'm using in [`raam`](https://raam.joebell.co.uk/) too and works rather nicely (although it needs a tidy up)